src/Tasks/Copy.cs

```diff
@@ -361,6 +391,42 @@ private void LogAlwaysRetryDiagnosticFromResources(string messageResourceName, p
             return true;
         }
 
+        private void RunLsof()
+        {
+            try
+            {
+                using (Process process = new Process())
+                {
+                    process.StartInfo.FileName = "lsof";
+
+                    process.StartInfo.RedirectStandardOutput = true;
+                    process.StartInfo.RedirectStandardError = true;
+                    process.StartInfo.UseShellExecute = false;
+                    process.StartInfo.CreateNoWindow = true;
+
+                    process.Start();
+
+                    string output = process.StandardOutput.ReadToEnd();
+                    string error = process.StandardError.ReadToEnd();
+
+                    process.WaitForExit();
+                    if (!string.IsNullOrEmpty(output))
```

Comment:
I'm not sure that there's a better answer for this part, but do not that this may rearrange the output and the error.